internal/agent/device/device.go

```diff
@@ -106,15 +109,26 @@ func NewAgent(
 
 // Run starts the device agent reconciliation loop.
 func (a *Agent) Run(ctx context.Context) error {
-	// orchestrates periodic fetching of device specs and pushing status updates
-	engine := NewEngine(
-		a.fetchSpecInterval,
-		a.syncDeviceSpec,
-		a.statusUpdateInterval,
-		a.statusUpdate,
+	var wg sync.WaitGroup
+	wg.Add(3)
+
+	// Goroutine to handle consoles synchronization
+	go a.syncConsoles(ctx, &wg)
+
+	// Goroutine to handle spec notifier which reads rendered devices from the server
+	notifierEngine := NewEngine(
+		NewUnit(time.Duration(a.fetchSpecInterval), a.specNotifier.PollAndNotify),
 	)
+	go notifierEngine.Run(ctx, &wg)
 
-	return engine.Run(ctx)
+	// Engine to handle device spec and status updates
+	specEngine := NewEngine(
+		NewUnit(time.Duration(a.fetchSpecInterval), a.syncDeviceSpec),
```

Comment:
question: Why are we making this generic if it is unnecessary for this change? I believe we can leave the existing sync logic in place, including the engine. My concern is this gives the impression that other things besides spec and status should be part of this loop. Which if anything we would be reducing down to one before we extend,

Reply:
The notifier runs with an engine.  It is possible to create a dedicated loop instead.

Reply:
yeah or move the engine out of agent. if we can keep this logic the same for this pr I think it will help with risk.

Reply:
Reverted the engine